shell/src/main/java/org/apache/accumulo/shell/commands/FateCommand.java

```diff
@@ -130,107 +132,131 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
-        filterTxid = new HashSet<>(args.length);
-        for (int i = 1; i < args.length; i++) {
-          try {
-            Long val = parseTxid(args[i]);
-            filterTxid.add(val);
-          } catch (NumberFormatException nfe) {
-            // Failed to parse, will exit instead of displaying everything since the intention was
-            // to potentially filter some data
-            System.out.printf("Invalid transaction ID format: %s%n", args[i]);
-            return 1;
-          }
+    }
+
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapter(ReadOnlyRepo.class, new InterfaceSerializer<>())
+        .registerTypeAdapter(Repo.class, new InterfaceSerializer<>())
+        .registerTypeAdapter(byte[].class, new ByteArraySerializer()).setPrettyPrinting().create();
+
+    List<FateStack> txStacks = new ArrayList<>();
+    for (Long txid : txids) {
+      List<ReadOnlyRepo<FateCommand>> repoStack = zs.getStack(txid);
+      txStacks.add(new FateStack(txid, repoStack));
+    }
+
+    return gson.toJson(txStacks);
+  }
+
+  private void printTx(Shell shellState, AdminUtil<FateCommand> admin, ZooStore<FateCommand> zs,
+      ZooReaderWriter zk, ServiceLock.ServiceLockPath tableLocksPath, String[] args, CommandLine cl,
+      boolean printStatus) throws InterruptedException, KeeperException, IOException {
+    // Parse transaction ID filters for print display
+    Set<Long> filterTxid = null;
+    if (args.length >= 2) {
+      filterTxid = new HashSet<>(args.length);
+      for (int i = 1; i < args.length; i++) {
+        try {
+          Long val = parseTxid(args[i]);
+          filterTxid.add(val);
+        } catch (NumberFormatException nfe) {
+          // Failed to parse, will exit instead of displaying everything since the intention was
+          // to potentially filter some data
+          throw new RuntimeException("Invalid transaction ID format: " + args[i], nfe);
         }
       }
+    }
 
-      // Parse TStatus filters for print display
-      EnumSet<TStatus> filterStatus = null;
-      if (cl.hasOption(statusOption.getOpt())) {
-        filterStatus = EnumSet.noneOf(TStatus.class);
-        String[] tstat = cl.getOptionValues(statusOption.getOpt());
-        for (String element : tstat) {
-          try {
-            filterStatus.add(TStatus.valueOf(element));
-          } catch (IllegalArgumentException iae) {
-            System.out.printf("Invalid transaction status name: %s%n", element);
-            return 1;
-          }
+    // Parse TStatus filters for print display
+    EnumSet<TStatus> filterStatus = null;
+    if (printStatus) {
+      filterStatus = EnumSet.noneOf(TStatus.class);
+      String[] tstat = cl.getOptionValues(statusOption.getOpt());
+      for (String element : tstat) {
+        try {
+          filterStatus.add(TStatus.valueOf(element));
+        } catch (IllegalArgumentException iae) {
+          throw new RuntimeException("Invalid transaction status name: " + element, iae);
```

Comment:
This, and in other places, you're converting a specific RTE to a more generic RTE. I think there's more value in keeping specific RTEs.

Additionally, the intent of the previous code looks like it was trying to communicate with the user by translating a well-understood exception into a message that would be helpful to the user. This new code rethrows the exception, and depending on the local log4j settings, the user may not even see it in the console. If they do see it, they'll see a very verbose stack trace that's probably unnecessary for communicating the error in the malformed transaction status name.